src/pudl/metadata/resources/sec10k.py

```diff
@@ -39,7 +39,105 @@
         "field_namespace": "sec10k",
     },
     "core_sec10k__quarterly_company_information": {
-        "description": "Company information extracted from SEC 10k filings.",
+        "description": (
+            """Company information extracted from SEC10k filings.
+This table provides attributes about SEC 10k filing companies across time.
+It represents a pivoted version of the raw company information table with extracted
+field values from the raw table as columns in this core table."""
+        ),
+        "schema": {
+            "fields": [
+                "central_index_key",
+                "report_date",
+                "filename_sec10k",
+                "phone_number",
+                "city",
+                "company_name",
+                "name_change_date",
+                "film_number",
+                "fiscal_year_end",
+                "sec10k_version",
+                "company_name_former",
+                "company_id_irs",
+                "organization_name",
+                "sec_act",
+                "sec_file_number",
+                "industry_id_sic",
+                "state",
+                "state_of_incorporation",
+                "street_address",
+                "address_2",
+                "zip_code",
+            ],
+            "primary_key": ["central_index_key", "report_date"],
+        },
+        "sources": ["sec10k"],
+        "etl_group": "pudl_models",
+        "field_namespace": "sec10k",
+    },
+    "core_sec10k__changelog_company_name": {
+        "description": (
+            """A record of SEC company name changes and the reported date of the name change.
+This table is pulled from ``core_sec10k__quarterly_company_information`` and contains
+data extracted from SEC 10k filings."""
+        ),
+        "schema": {
+            "fields": [
+                "central_index_key",
+                "report_date",
+                "company_name",
+                "name_change_date",
+                "company_name_former",
+            ],
+            "primary_key": ["central_index_key", "report_date"],
+        },
```

Comment:
Would adding a foreign key relationship to the core company information table just be on the `central_index_key` and `report_date`?

Reply:
Typically a FK constraint points to the PK columns in another table, so if those were the PKs in the company info table, that's probably be what you want.  But I've left comments both on the natural PK of the company info table, and on what a better normalized version of this table might look like.